### PR TITLE
Gui toolkits

### LINKED ISSUE #31: Please write about using a GUI toolkit [from sourceforge / feature requests]
Hello --
The documentation on using something like CLIM is
overwhelming. We need to simple tips on building GUIs,
simple examples.
Please choose a simple a Libre Software solution.
Thank you.

synthespian

[http://sourceforge.net/p/cl-cookbook/feature-requests/4/]

### PR DESCRIPTION
**Pull request by vindarel**
First impressions?

https://vindarel.github.io/cl-cookbook/gui.html

- introduction, presentations, structure, examples usefulness…?

fixes #31

### CODE REVIEW

**Comment by bendudson** [comment]
I've never tried writing a GUI in Lisp, but might have to try now: This looks really nice!

**Comment by phoe** [comment]
Amazing stuff. :heart:

File: gui.md [hunk 1]
```diff
@@ -0,0 +1,917 @@
...
+Qt is free for open-source software, however you'll want to check the
+conditions to ship proprietary ones.
+
+The [Qtools][qtools] bindings target Qt4. The Qt5 Lisp bindings are
+yet to be created.
+
+<!-- possible future: gobject-introspection -->
+
+- **Framework written in**: C++
+- **Portability**: multiplatform, Android, embedded systems, WASM.
```
Comment (Shinmera) [inline gui.md]: Qtools only runs on x86 desktop platforms on windows, macos, and linux.

File: gui.md [hunk 2]
```diff
@@ -0,0 +1,917 @@
...
+
+- **Bindings documentation**: lengthy explanations, a few examples. Prior Qt knowledge is required.
+- **Bindings stability**: stable
+- **Bindings activity**: active
+- **Qt Licence**: both commercial and open source licences.
+- Example applications:
+  - https://github.com/Shinmera/qtools/tree/master/examples
+  - https://github.com/Shirakumo/lionchat
+  - https://github.com/shinmera/halftone - a simple image viewer
+  - https://github.com/shirakumo/cl-gamepad
```
Comment (Shinmera) [inline gui.md]: The visualiser is no longer part of cl-gamepad.

File: gui.md [hunk 3]
```diff
@@ -0,0 +1,917 @@
...
+(define-subwidget (main-window layout) (q+:make-qhboxlayout main-window)
+  (q+:add-widget layout name)
+  (q+:add-widget layout go-button))
+~~~
+
+and we show them:
+
+~~~lisp
+(with-main-window
+  (make-instance 'main-window))
```
Comment (Shinmera) [inline gui.md]: This should be: `(with-main-window (window 'main-window))`

File: gui.md [hunk 4]
```diff
@@ -0,0 +1,917 @@
...
+  (signal! main-window (name-set string) (q+:text name)))
+~~~
+
+You can run again
+
+~~~lisp
+(with-main-window
+  (make-instance 'main-window))
+~~~
+
```
Comment (Shinmera) [inline gui.md]: You should mention that it is not necessary to create an extra signal every time. You could just as well pack the code from the name-set slot into the go-button slot. In fact, most of the time you won't create extra signals of your own.

**Comment by Shinmera** [comment]
This is pretty good. I'm just sad that Alloy can't be included here yet...

**Comment by vindarel** [comment]
Thank you for the review!

File: gui.md (hunk 4 above)
Comment (vindarel) [reply gui.md]: got it, thanks.

**Pull request merged**